addons/pt/saft/item.go

```diff
@@ -0,0 +1,50 @@
+package saft
+
+import (
+	"github.com/invopop/gobl/org"
+	"github.com/invopop/gobl/tax"
+	"github.com/invopop/validation"
+)
+
+func validateItem(item *org.Item) error {
+	if item == nil {
+		return nil
+	}
+
+	return validation.ValidateStruct(item,
+		validation.Field(&item.Unit, validation.Required),
+		validation.Field(&item.Ext,
+			tax.ExtensionsRequire(ExtKeyProductType),
+			validation.Skip,
+		),
+	)
+}
+
+func normalizeItem(item *org.Item) {
+	setDefaultUnit(item)
+	setDefaultProductType(item)
+}
+
+func setDefaultProductType(item *org.Item) {
+	if item == nil {
+		return
+	}
+
+	if item.Ext == nil {
+		item.Ext = make(tax.Extensions)
+	}
+
+	if _, ok := item.Ext[ExtKeyProductType]; !ok {
+		item.Ext[ExtKeyProductType] = ProductTypeService
```

Comment:
Is it worth trying to map this from the unit?

Reply:
Interesting idea. Given that the unit is mandatory, we can definitely make a better guess about the right product type considering the unit. I'll give it a try.

Reply:
Addressed in 0b756c2893e4f63cc595a21eeb7bf376392a4ca2